lib/ts/recipe/multifactorauth/types.ts

```diff
@@ -0,0 +1,167 @@
+import { GeneralErrorResponse } from "../../types";
+import RecipeUserId from "../../recipeUserId";
+import { SessionContainer } from "../session";
+import { SessionContainerInterface } from "../session/types";
+
+export type MFARequirement =
+    | {
+          id: string;
+          maxAgeInSeconds?: number;
+      }
+    | string;
+
+export type MFARequirementList = (
+    | {
+          oneOf: MFARequirement[];
+      }
+    | {
+          allOf: MFARequirement[];
+      }
+    | MFARequirement
+)[];
+
+export type MFAClaimValue = {
+    c: Record<string, number>;
+    n: string[];
+};
+
+export type TypeInput = {
+    firstFactors?: string[];
+
+    getGlobalMFARequirements?: (
+        userId: string,
+        recipeUserId: RecipeUserId,
+        tenantId: string,
+        session: SessionContainer | undefined,
+        factorsSetUpByTheUser: string[],
+        enabledFactors: string[],
+        enabledFactorsForTenant: string[],
+        completedFactors: Record<string, number>,
+        userContext: any
+    ) => Promise<MFARequirementList> | MFARequirementList;
+    getMFARequirementsForFactorSetup?: (
+        factorId: string,
+        session: SessionContainer,
+        factorsSetUpByTheUser: string[],
+        enabledFactors: string[],
+        enabledFactorsForTenant: string[],
+        completedFactors: Record<string, number>,
+        userContext: any
+    ) => Promise<MFARequirementList> | MFARequirementList;
+
+    override?: {
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis?: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type TypeNormalisedInput = {
+    firstFactors?: string[];
+
+    getGlobalMFARequirements: (
+        userId: string,
+        recipeUserId: RecipeUserId,
+        tenantId: string,
+        session: SessionContainer | undefined,
+        factorsSetUpByTheUser: string[],
+        enabledFactors: string[],
+        enabledFactorsForTenant: string[],
+        completedFactors: Record<string, number>,
+        userContext: any
+    ) => Promise<MFARequirementList> | MFARequirementList;
+    getMFARequirementsForFactorSetup: (
+        factorId: string,
+        session: SessionContainer,
+        factorsSetUpByTheUser: string[],
+        enabledFactors: string[],
+        enabledFactorsForTenant: string[],
+        completedFactors: Record<string, number>,
+        userContext: any
+    ) => Promise<MFARequirementList> | MFARequirementList;
+
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
```

Comment:
Instead of having factor id for passwordless where we have "otp-mobile" etc.. we just have it as "passwordless", and in passwordless init, we define:
```
passwordless.init({
	contact_method: “…”,
	flow_type: “..”,
	mfa: {
		contact_method: “…”
	}
})
oneOf: […]
```
- The `mfa` object doesn't have flow type cause we don't support magic link